// react-native-get-music-files/lib/module/index.js
import { NativeModules, Platform } from 'react-native';
const LINKING_ERROR = `The package 'react-native-music-files' doesn't seem to be linked. Make sure: \n\n` + Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo Go\n';

// @ts-expect-error
const isTurboModuleEnabled = global.__turboModuleProxy != null;
const TurboSongsModule = isTurboModuleEnabled ? require('./NativeTurboSongs').default : NativeModules.TurboSongs;
const TurboSongs = TurboSongsModule ? TurboSongsModule : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
var SortSongOrder = /*#__PURE__*/function (SortSongOrder) {
  SortSongOrder["ASC"] = "ASC";
  SortSongOrder["DESC"] = "DESC";
  return SortSongOrder;
}(SortSongOrder || {});
var SortSongFields = /*#__PURE__*/function (SortSongFields) {
  SortSongFields["TITLE"] = "TITLE";
  SortSongFields["DURATION"] = "DURATION";
  SortSongFields["ARTIST"] = "ARTIST";
  SortSongFields["GENRE"] = "GENRE";
  SortSongFields["ALBUM"] = "ALBUM";
  return SortSongFields;
}(SortSongFields || {});
const getAll = async options => {
  try {
    return await TurboSongs.getAll(options);
  } catch (e) {
    return `${e}`;
  }
};
const getAlbums = async options => {
  try {
    return await TurboSongs.getAlbums(options);
  } catch (e) {
    return `${e}`;
  }
};
const searchSongs = async options => {
  try {
    return await TurboSongs.search(options);
  } catch (e) {
    return `${e}`;
  }
};
export { getAll, getAlbums, searchSongs, SortSongFields, SortSongOrder };
